observer/probers/http/http.go

```diff
@@ -37,9 +40,14 @@ func (p HTTPProbe) isExpected(received int) bool {
 // Probe performs the configured HTTP request.
 func (p HTTPProbe) Probe(timeout time.Duration) (bool, time.Duration) {
 	client := http.Client{Timeout: timeout}
+	req, err := http.NewRequest("GET", p.url, nil)
+	if err != nil {
+		return false, 0
+	}
+	req.Header.Set("User-Agent", p.useragent)
```

Comment:
Probably better to leave the default in place than to override it with the empty string, in case the config field isn't set.

Reply:
I opted to set a custom User-Agent "letsencrypt/boulder_observer_client" if none is set. I think that is a better approach to enable server operators to trace requests. This makes it harder for implementations to use the generic net/http default User-Agent.